CATTY-235 Add tests for VariableContainer (part 5)

DhrubojyotiBis1 opened the pull request:
### Tests added for the following methods of VariableContainer:

- isEqualToVariablesContainer
- mutableCopy

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Verify that the Jira ticket is in the status *Ready for Development*
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

m-herold commented on src/CattyTests/DataModel/VariablesContainerTest.swift:
```diff
@@ -236,4 +236,41 @@ final class VariablesContainerTest: XCTestCase {
         XCTAssertTrue(lists[1].isList)
     }
+
+    func testIsEqualToVariablesContainer() {
```
Please add a few other test methods to verify all possible code execution paths of this complex method (e.g. unequal number of variables/lists, variable with same name but different object, variable with same name but different values, ...).

m-herold commented on src/CattyTests/DataModel/VariablesContainerTest.swift:
```diff
@@ -236,4 +236,41 @@ final class VariablesContainerTest: XCTestCase {
+        let copyContainer = container.mutableCopy() as! VariablesContainer
+
+        XCTAssertTrue(container.isEqual(to: copyContainer))
+        XCTAssertFalse(container == copyContainer)
```
Please also check whether the items of the container have the same reference or not.

DhrubojyotiBis1 commented:
Hi, @m-herold  I have made the changes. please review

DhrubojyotiBis1 commented:
Hi, @m-herold  I have removed all the merge conflicts and the swiftLint warning please review.

m-herold commented on src/CattyTests/DataModel/VariablesContainerTest.swift:
```diff
@@ -234,6 +234,121 @@ final class VariablesContainerTest: XCTestCase {
+        XCTAssertTrue(container.isEqual(to: copyContainer))
+        XCTAssertFalse(container == copyContainer)
+
+        container.change(variable, byValue: 10)
```
Does that implicitly mean that there is no deep copy of the items of the container (i.e. `list` and `variable`)? If so, please better remove the variable change and replace it with pseudo code: `container.list has same reference than copyContainer.list `(and the same with `variable`).

DhrubojyotiBis1 replied:
Yes, there is no deep copy of the items of the container.
sorry I couldn't understand, are you asking to write the pseudo-code: `container.list has same reference than copyContainer.list` in comment which replaces `container.change(variable, byValue: 10)`

m-herold replied:
Yes please replace this, but not with a comment, but with actual code (i.e. `XCTAssertTrue(container.allVariables()[0] === copyContainer.allVariables()[0])` or similar).

DhrubojyotiBis1 replied:
Sorry, I could understand it😛, I will make the changes.

DhrubojyotiBis1 commented:
> Please rebase instead of merging the develop into your branch and/or squash your commits so that there are no _Merge branch 'develop' into CATTY-..._ commit messages in the develop's commit history (those do not contain any information and will just make the history less readable).

Hi, @m-herold I rebased and squashed as you suggested but after removing merge conflicts locally when I push the changes I get the same merge conflict again on GitHub, so I resolved that conflict on GitHub itself.
Please correct me if I am doing something wrong.

DhrubojyotiBis1 commented:
@m-herold I have rebased and squashed, please review.
